packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -2131,9 +2131,18 @@ class Navigator extends StatefulWidget {
     bool rootNavigator = false,
     bool nullOk = false,
   }) {
-    final NavigatorState navigator = rootNavigator
-        ? context.findRootAncestorStateOfType<NavigatorState>()
-        : context.findAncestorStateOfType<NavigatorState>();
+    // Handles the case where the input context is a navigator element.
```

Comment:
/cc @gspencergoog I seem to remember that you recently also ran into the problem that "self" is not included when you look up ancestors. How did we solve the problem there again?

Reply:
I created [`_getAncestor`](https://cs.opensource.google/flutter/flutter/+/master:packages/flutter/lib/src/widgets/focus_traversal.dart;l=26?q=_getAncestor&ss=flutter%2Fflutter) to fetch the parent so that I can start searching using `getElementForInheritedWidgetOfExactType` from the parent element, which avoids returning itself just by starting the search with the parent element.

Reply:
And my problem was the opposite: `getElementForInheritedWidgetOfExactType` can return self, and I didn't want it to.